### DISCUSSION TITLE
ORCID Social Identity Provider community extension

### DISCUSSION THREAD

**Discussion by cgeorgilakis** [keycloak/keycloak#8620] [Archive]
Keycloak does not support [ORCID ](https://orcid.org/) as a Social Identity Provider. ORCID is commonly used by researchers as a sign-in method. Obtaining an authenticated ORCID identifier also allows requesting permission to use the ORCID API for reading or updating the user’s ORCID record which contains professional activities, publications and affiliations. 
Although ORCID can be configured as an OpenID Connect Identity Provider, the current ORCID OP implementation does not support the release of email and other information (see ORCID OpenID Connect documentation).

We believe that it will be usefull Keycloak supporting ORCID as Social Identity Provider. That's why we made a [PR](https://github.com/keycloak/keycloak/pull/8618) for this. Details can be found in the [jira issue](https://issues.redhat.com/browse/KEYCLOAK-19595).

I have made implementation for SocialLoginTest with an appropriate comment in code ( ORCID registry supports only https, fields visibility).

**Comment by stianst** [comment]
I don't think this is something we want in Keycloak directly, see my comment on the PR for more details

**Comment by cgeorgilakis** [comment]
In the repository https://github.com/eosc-kc/keycloak-orcid I have created the Social ORCID Identity Provider as @stianst wants in the [PR discussion](https://github.com/keycloak/keycloak/pull/8618). It is working well with current Keycloak version 15.0.2. Documentation is provided in readme.md file.
What is the procedure in order to become official Keycloak extension?
As I have mentioned in the PR, I do not know how I can implement SocialLoginTest in order to have tests for ORCID Social Identity Provider.

**Reply by stianst** [reply]
For an extension it's up to you how to test it, so you don't need to use the SocialLoginTest.

To get it added as an extension send a PR to https://github.com/keycloak/keycloak-web/tree/main/extensions

**Reply by cgeorgilakis** [reply]
PR : https://github.com/keycloak/keycloak-web/pull/191
